src/backend/cuda/kernel/join.hpp

```diff
@@ -73,6 +73,15 @@ namespace cuda
                         blocksPerMatY * X.dims[3],
                         1);
 
+            int maxBlocksY = cuda::getDeviceProp(cuda::getActiveDeviceId()).maxGridSize[1];
+            // Increase the work per block until the condition is satisfied
+            // Increase threads only upto 512
+            while(blocks.y > maxBlocksY) {
```

Comment:
I made the PR description to suggest better ways to do this. Right now, this is the simplest way to get rid of this bug. We definitely need a more permanent solution for this. But until we figure out what that is, this will work.